src/cpp/include/openvino/genai/rag/text_embedding_pipeline.hpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <filesystem>
+#include <optional>
+#include <variant>
+
+#include "openvino/genai/tokenizer.hpp"
+
+namespace ov {
+namespace genai {
+
+using EmbeddingResult = std::variant<std::vector<float>, std::vector<int8_t>, std::vector<uint8_t>>;
+
+class OPENVINO_GENAI_EXPORTS TextEmbeddingPipeline {
+public:
+    enum class OPENVINO_GENAI_EXPORTS PoolingType {
+        /**
+         * @brief First token embeddings
+         */
+        CLS = 0,
+        /**
+         * @brief The average of all token embeddings
+         */
+        MEAN = 1,
+    };
+
+    struct OPENVINO_GENAI_EXPORTS Config {
+        std::optional<size_t> max_length;
+        PoolingType pooling_type = PoolingType::CLS;
+        bool normalize = false;
+        std::optional<std::string> query_instruction;
+        std::optional<std::string> embed_instruction;
+
+        Config() = default;
+        explicit Config(const ov::AnyMap& properties);
+    };
+
+    /**
+     * @brief Constructs a pipeline from xml/bin files, tokenizer and configuration in the same dir.
+     *
+     * @param models_path Path to the dir model xml/bin files, tokenizer
+     * @param device optional device
+     * @param plugin_config optional plugin_config properties
+     */
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          const std::optional<Config>& config = std::nullopt,
```

Comment:
do we need `std::optional` here? I suppose instead of `TextEmbeddingPipeline(path, device, std::nullopt, properties)` users can use the ctor below `TextEmbeddingPipeline(path, device, properties)` which don't accept `Config`

Reply:
We need `std::optional` here to support creation with no config or properties. Current ctors supports next calls:
```c++
    ov::genai::TextEmbeddingPipeline pipeline(models_path, device);
    ov::genai::TextEmbeddingPipeline pipeline(models_path, device, config);
    ov::genai::TextEmbeddingPipeline pipeline(models_path,
                                               device,
                                               ov::genai::normalize(true),
                                               ov::cache_dir(".ov_cache"));
```

Reply:
discussed offline